wagtail/blocks/struct_block.py

```diff
@@ -17,24 +17,36 @@
 
 
 class StructBlockValidationError(ValidationError):
-    def __init__(self, block_errors=None):
+    def __init__(self, block_errors=None, non_block_errors=None):
         self.block_errors = block_errors
-        super().__init__("Validation error in StructBlock", params=block_errors)
+        self.non_block_errors = non_block_errors
+
+        params = {}
+        if block_errors:
+            params.update(block_errors)
+        if non_block_errors:
+            params[NON_FIELD_ERRORS] = non_block_errors
+        super().__init__("Validation error in StructBlock", params=params)
 
 
 class StructBlockValidationErrorAdapter(Adapter):
     js_constructor = "wagtail.blocks.StructBlockValidationError"
 
     def js_args(self, error):
-        if error.block_errors is None:
-            return [None]
+        if error.non_block_errors:
```

Comment:
I wonder why we haven't been using `foo_errors.as_data()` directly (without explicit truthy `if` checks) in this method, like how we do it in `StreamBlock`:

https://github.com/wagtail/wagtail/blob/dc2afe89b94265ea1dd592ea0e64090b05de2cf8/wagtail/blocks/stream_block.py#L39-L49

Is it because these `foo_errors` can be `None` instead of empty `ErrorList`s?